test/test_list_layout.cpp

```diff
@@ -0,0 +1,279 @@
+            }
+    
+        }
+        SUBCASE("list<list<int>>")
+        {
+            auto d = test::iota_vector<T>(28);
+            std::vector<std::vector<std::vector<T>>> values = {
+                {
+                    {d[0],d[1], d[2], d[3]}, 
+                    {d[4], d[5], d[6]},
+                    {d[7], d[8], d[9], d[10]}
+                },
+                {
+                    {d[11],d[12], d[13], d[14]}, 
+                    {d[15], d[16]}, 
+                    {d[17], d[18], d[19], d[20], d[21]}
+                },
+                {
+                    {d[22],d[23], d[24], d[25]}, 
+                    {d[26], d[27]}
+                }
+            };
+
+            auto outer_list_array_data = test::make_array_data_for_list_of_list_of_scalars(values);
+        
+            using data_storage = sparrow::array_data;
+            using inner_layout_type = sparrow::fixed_size_layout<T, data_storage>;
+            using inner_list_layout_type = sparrow::list_layout<inner_layout_type, data_storage,  std::int64_t>;
+            using outer_list_layout_type = sparrow::list_layout<inner_list_layout_type, data_storage,  std::int64_t>;
+            
+            SUBCASE("operator[]")
+            {
+                outer_list_layout_type outer_list_layout(outer_list_array_data);
+                CHECK_EQ(outer_list_layout.size(), values.size());
+                for(std::size_t i = 0; i < values.size(); i++)
+                {
+                    auto maybe_list = outer_list_layout[i];
+                    CHECK_EQ(maybe_list.has_value(), true);
+                    auto list = maybe_list.value();
+                    CHECK_EQ(list.size(), values[i].size());
+                    for(std::size_t j = 0; j < values[i].size(); j++)
+                    {
+                        auto maybe_inner_list = list[j];
+                        CHECK_EQ(maybe_inner_list.has_value(), true);
+                        auto inner_list = maybe_inner_list.value();
+                        CHECK_EQ(inner_list.size(), values[i][j].size());
+                        for(std::size_t k = 0; k < values[i][j].size(); k++)
+                        {
+                            auto maybe_value = inner_list[k];
+                            CHECK_EQ(maybe_value.has_value(), true);
+                            CHECK_EQ(maybe_value.value(), values[i][j][k]);
+                        }
+                    }
+                }
+            }
+            SUBCASE("values")
+            {
+                SUBCASE("const")
+                {
+                    outer_list_layout_type outer_list_layout(outer_list_array_data);
+                    const outer_list_layout_type& const_outer_list_layout = outer_list_layout;
+                    for(auto list : const_outer_list_layout.values())
+                    {
+                        for(auto elemets  : list)
+                        {
+                            CHECK_EQ(elemets.has_value(), true);
```

Comment:
Can you check the value too ?

Reply:
lets not...lets rather write one **generic** consistency test for all layouts which checks of value() / bitmap() and normal iterators are consistent with each other.